// The Firmament project
// Copyright (c) 2011-2012 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// RegistrationMessage informs about the existence of a resource.

package firmament;

import "base/resource_desc.proto";
import "base/resource_topology_node_desc.proto";

message RegistrationMessage {
  required string uuid = 1;
  optional string location = 2;
  optional ResourceDescriptor res_desc = 3;
  optional ResourceTopologyNodeDescriptor rtn_desc = 4;
  optional string hostname = 5;
}
